ui/jquery.ui.mouse.js

```diff
@@ -23,10 +23,117 @@
 //JASPERSOFT #1 #3
 define(function(require) {
 
-	var $ = require("./jquery.ui.widget");
-	var isIE = function () {
-		return navigator.appName === "Microsoft Internet Explorer" || navigator.userAgent.indexOf("Trident/") >= 0;
-	};
+    var $ = require("./jquery.ui.widget");
+
+    var isIE = function () {
+        return navigator.appName === "Microsoft Internet Explorer" || navigator.userAgent.indexOf("Trident/") >= 0;
+    };
+
+    var detectedIEVersion = false;
+
+    var getIEVersion = function () {
+        var version = 0;
+        if (detectedIEVersion !== false) {
+            return detectedIEVersion;
+        }
+
+        if (this.isIE()) {
+            if (navigator.appName === "Netscape") {
+                var ua = navigator.userAgent;
+                var re = new RegExp("Trident/.*rv:([0-9]{1,}[\\.0-9]{0,})");
+                if (re.exec(ua) != null) {
+                    version = parseFloat(RegExp.$1);
+                }
+            } else {
+                var msVersion = navigator.appVersion.split("MSIE")[1];
+                version = parseFloat(msVersion);
+            }
+        }
+
+        detectedIEVersion = version;
+        return version;
+    };
+
+    var isIE11 = function () {
+        return this.getIEVersion() === 11;
+    };
+    var RECT = function () {
```

Comment:
It may be better if this were a function that returns an object instead of a function, as it seems to be used as an object later on. This is just cosmetic and doesn't affect the fix.  Maybe note for later.

Reply:
right, agree, thanks